### PR TITLE
Fix some more potential memory issues

### PR DESCRIPTION
**Pull request by ashie**
Please be careful to merge it, I'm not so sure about both these Windows API & Ruby  C extension.
(All tests are passed on my environment though.)
Please check these patches one by one.

### CODE REVIEW

**Review by cosmo0920 — APPROVED** [review]
I've confirmed that these changes make sense and working with Fluentd Windows certstore integration.

File: ext/certstore/loader.c [hunk 1]
```diff
@@ -255,7 +255,7 @@ rb_win_certstore_loader_find_certificate(VALUE self, VALUE rb_thumbprint)
   blob.cbData = CERT_THUMBPRINT_SIZE;
   blob.pbData = pbThumb;
   CryptStringToBinaryW(winThumbprint,
-                       CERT_THUMBPRINT_STR_LENGTH,
+                       0,
```
Comment (cosmo0920) [inline ext/certstore/loader.c:258]: Oh, this parameter should be for C++ containers! :open_mouth:

File: ext/certstore/loader.c [hunk 2]
```diff
@@ -428,7 +427,7 @@ rb_win_certstore_loader_export_pfx(VALUE self, VALUE rb_thumbprint, VALUE rb_pas
                                         pContext);
   if (!pContext) {
     _snprintf_s(errBuf,
-                256,
+                sizeof(errBuf),
```
Comment (cosmo0920) [inline ext/certstore/loader.c:430]: nit: This parameter should be byte size or element count?

byte size: sizeof
element count: [_countof](https://docs.microsoft.com/ja-jp/cpp/c-runtime-library/reference/countof-macro?view=vs-2019)

Comment (ashie) [reply ext/certstore/loader.c:430]: Results of them are same in multi byte string.
`char` (a.k.a `CHAR` in Win32) is 1 byte, it's C lang's spec.

**Pull request merged**